[cydtseng] iP

cydtseng opened the pull request:
# Duke Project

> *Chatbots represent a new trend in how people access information, make decisions and communicate.”  — Christie Pitts ( Verizon Ventures)*

The Duke Project is a chatbot based todo list with the following features
- Command-based input
- Adding events, deadlines and todos
- Showing list of tasks to do
- All tasks saved locally

How to run:
1. Download the release [here](https://github.com/cydtseng/ip/releases/tag/A-Jar).
2. Run the jar file.
3. Start adding tasks.
4. Let Duke manage tasks for you! :partying_face:

**Interactivity progress:**

- [x] Text based UI
- [ ] GUI with JavaFX 

---
Entry point `main` of Duke program.
```java
public static void main(String[] args) {
    new Duke().run();
}
```

Commit:
Add Gradle support

cydtseng committed:
implement duke level 1, greet, echo and exit.

cydtseng committed:
implement level 2, add, list

cydtseng committed:
add Task class

cydtseng committed:
implement Level 3, Mark as Done

cydtseng committed:
add and define Deadline class

cydtseng committed:
add and define Event class

cydtseng committed:
add and define Todo class

cydtseng committed:
remove getter for isDone variable from task class

cydtseng committed:
modify Duke class to make use of subclasses from Task class

cydtseng committed:
add DukeException class

cydtseng committed:
modify Duke class to handle exceptions

cydtseng committed:
add static deleteTask method for Task class

cydtseng committed:
implement level 6 delete functionality

cydtseng committed:
add enum for all possible cmds

cydtseng committed:
add basic input text and expected text ui

cydtseng committed:
Update gitignore to ignore data folder

cydtseng committed:
Add additional constructor for Deadline class

cydtseng committed:
Add additional constructor for Event class

cydtseng committed:
Add additional constructor for Task class

cydtseng committed:
Add additional constructor for Todo class

cydtseng committed:
Implement file based task storage for Duke class

cydtseng committed:
Change access modifier for variables in Task class

cydtseng committed:
Change constructor and toString method for Deadline class

cydtseng committed:
Change constructor and toString method for Event class

cydtseng committed:
Add parsing for date and time for deadline and event commands in Duke class

cydtseng committed:
Merge branch 'branch-Level-7'

cydtseng committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

cydtseng committed:
Refactor user command processing to Parser class

cydtseng committed:
Refactor loading of tasks from file and saving of tasks to Storage class

cydtseng committed:
Refactor interactions with user to Ui class

cydtseng committed:
Refactor task list operations to TaskList class

cydtseng committed:
Remove unnecessary variable and methods from Task class

cydtseng committed:
Apply more OOP to Duke class using newly created classes: Parser, Storage, TaskList and Ui

cydtseng committed:
Add all classes to duke package

cydtseng committed:
Merge remote-tracking branch 'origin/add-gradle-support'

cydtseng committed:
Add JUnit tests for Parser and Todo

cydtseng committed:
Update gitignore to ignore META-INF

cydtseng committed:
Add javadoc comments for roughly half of all non-private classes/methods

cydtseng committed:
Tweak code to comply with coding standard

cydtseng committed:
Add find functionality that allows users to find a task by searching for a keyword

cydtseng committed:
Merge branch 'branch-A-JavaDoc'

cydtseng committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Event.java
#	src/main/java/duke/Todo.java
#	src/main/java/duke/Ui.java

cydtseng committed:
Resolve bug where new tasks are not written to save file

cydtseng committed:
add GUI as main interaction method for Duke

cydtseng committed:
add checkstyle files

Jayanth-Balasubramanian commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,127 @@
+
+    public static void saveTasksToFile(ArrayList<Task> taskList) {
+        try {
+            String toWrite = "";
```
A little pedantic, but it might be better to use a noun for this String, such as tasksToWrite

Jayanth-Balasubramanian commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,20 @@
+    public void testParseByeCommand(){
+        Assertions.assertTrue(parser.isByeCommand("bye"));
+        Assertions.assertTrue(parser.isByeCommand("bye "));
+        Assertions.assertTrue(parser.isByeCommand(" bye"));
```
It's great that you're testing these edge cases!

Jayanth-Balasubramanian commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,301 @@
+public class Parser {
+    private final Stage stage;
+
+    public enum CmdType {
```
enum constants should be in SCREAMING_SNAKE_CASE

Jayanth-Balasubramanian commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,301 @@
+        return userInput.split(" ")[0].equals(CmdType.find.name());
+    }
+
+    public String parseCommandWithResponse(String userInput) {
```
This method seems way too long. As a suggestion, perhaps you could break your method up into handleByeCommand, handleMarkCommand etc so it's easier to debug.

Jayanth-Balasubramanian reviewed (commented):
LGTM except some minor nitpicks. Very clean code!

cydtseng committed:
Update project with assertions enabled for intellij and gradle

For now, assertions have been added to the DialogBox class and the
Storage class.

The assertions are to check assumptions of non null imageviews and
dialogbox labels as well as storage file directory and saved task
file existence.

This was referenced in enabling assertions
https://se-education.org/guides/tutorials/intellijUsefulSettings.html

cydtseng committed:
Refactor some code that does not adhere to SLAP

cydtseng committed:
Merge pull request #1 from cydtseng/branch-A-Assertions

Update project with assertions enabled for intellij and gradle

cydtseng committed:
Merge pull request #2 from cydtseng/branch-A-CodeQuality

Refactor some code that does not adhere to SLAP

cydtseng committed:
Add help command to show all possible command formats for users

cydtseng committed:
Merge pull request #4 from cydtseng/branch-C-Help

Add help command to show all possible command formats for users

cydtseng committed:
Add personality to Duke with new phrases, styling for
colors and fonts

cydtseng committed:
Merge branch 'branch-A-Personality'

cydtseng committed:
Add enhancement to GUI font and dialog borders

cydtseng committed:
Shift files into task package for better management

cydtseng committed:
Add more automated tests in JUnit

cydtseng committed:
Merge branch 'branch-A-MoreTesting'

cydtseng committed:
Refactor Ui code to adhere to SLAP

cydtseng committed:
Add logo for application

cydtseng committed:
Add Ui.png file for docs

cydtseng committed:
Add more detailed user guide for ip

cydtseng committed:
Update code to adhere to checkstyle
